test_infra/ess/readme.md

```diff
@@ -0,0 +1,22 @@
+## Terraform-managed Elastic Stack deployment for integration tests 
+
+### Required variables
+`stack_version` - The version of the Elastic Stack to deploy, e.g., `9.1.0-SNAPSHOT`.
+
+### Stable snapshots and images override
+
+Variables can be overridden to use stable snapshots or images instead of the latest snapshot. This is useful for testing against a specific version of the Elastic Stack.
+* `integration_server_docker_image` - The Docker image for the integration server
+* `elasticsearch_docker_image` - The Docker image for Elasticsearch
+* `kibana_docker_image` - The Docker image for Kibana
+
+#### The priority of the docker images is as follows:
+example for `integration_server_docker_image`:
+1) Variable override: `TF_VAR_integration_server_docker_image...` takes precedence over
+2) `pkg/testing/ess/create_deployment_csp_configuration.yaml` - `docker` section pins the images versions , e.g.:
+```yaml
+docker:
+  integration_server_docker_image: "docker.elastic.co/cloud-release/elastic-agent-cloud:9.1.0-SNAPSHOT"
+  elasticsearch_docker_image: "..."
+  kibana_docker_image: "..."
```

Comment:
and there should be a 3) here which is the stack version in case no `integration_server_docker_image` and no explicit versions are specified in the yaml 🙂